src/cli/project.ts

```diff
@@ -0,0 +1,172 @@
+import { existsSync, readFileSync } from "node:fs";
+import path from "node:path";
+
+import { Command } from "@commander-js/extra-typings";
+import { confirm } from "@inquirer/prompts";
+
+import { findFileUpwards } from "cli/utils";
+import sindri from "lib";
+import { ApiError } from "lib/api";
+
+async function deleteProject(projectIdentifier: string): Promise<void> {
+  try {
+    await sindri._client.internal.projectDelete(projectIdentifier);
+    sindri.logger.debug(`Deleted project "${projectIdentifier}".`);
+  } catch (error) {
+    if (error instanceof ApiError && error.status === 401) {
+      sindri.logger.error(
+        "Your credentials are invalid. Please log in again with `sindri login`.",
+      );
+    } else if (error instanceof ApiError && error.status === 404) {
+      sindri.logger.error(`The project "${projectIdentifier}" does not exist.`);
+    } else {
+      sindri.logger.fatal("An unknown error occurred.");
+      sindri.logger.error(error);
+    }
+    return process.exit(1);
+  }
+}
+
+function findProjectName(directory: string): string {
+  // Find `sindri.json` and move into the root of the project directory.
+  const directoryPath = path.resolve(directory);
+  if (!existsSync(directoryPath)) {
+    sindri.logger.error(
+      `The "${directoryPath}" directory does not exist. Aborting.`,
+    );
+    return process.exit(1);
+  }
+  const sindriJsonPath = findFileUpwards(/^sindri.json$/i, directoryPath);
+  if (!sindriJsonPath) {
+    sindri.logger.error(
+      `No "sindri.json" file was found in or above "${directoryPath}". Aborting.`,
+    );
+    return process.exit(1);
+  }
+  sindri.logger.debug(`Found "sindri.json" at "${sindriJsonPath}".`);
+  const rootDirectory = path.dirname(sindriJsonPath);
+  sindri.logger.debug(`Changing current directory to "${rootDirectory}".`);
+  process.chdir(rootDirectory);
+
+  // Load `sindri.json`.
+  let sindriJson: object = {};
+  try {
+    const sindriJsonContent = readFileSync(sindriJsonPath, {
+      encoding: "utf-8",
+    });
+    sindriJson = JSON.parse(sindriJsonContent);
+    sindri.logger.debug(
+      `Successfully loaded "sindri.json" from "${sindriJsonPath}":`,
+    );
+    sindri.logger.debug(sindriJson);
+  } catch (error) {
+    sindri.logger.fatal(
+      `Error loading "${sindriJsonPath}", perhaps it is not valid JSON?`,
+    );
+    sindri.logger.error(error);
+    return process.exit(1);
+  }
+  if (!("name" in sindriJson) || typeof sindriJson.name !== "string") {
+    sindri.logger.error('No "name" field found in "sindri.json". Aborting.');
+    return process.exit(1);
+  }
+  return sindriJson.name;
+}
+
+async function getSomeProjectNames(teamName: string): Promise<string[]> {
```

Comment:
I don't feel passionate about this suggestion, but my first readthrough of the code misinterpreted why this function is here. I thought you were laying the scaffolding out for future `project` subcommands.  
```suggestion
async function getSomeProjectNames(teamName: string): Promise<string[]> {
  // Used as an efficient iterator for `projectDeleteCommand`.
```